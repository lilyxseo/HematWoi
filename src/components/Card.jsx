export default function Card({ title, actions, children }) {
  return (
    <div className="card">
      {(title || actions) && (
        <div className="flex items-center justify-between mb-2">
          {title && <h2 className="font-semibold">{title}</h2>}
          {actions}
        </div>
      )}
      {children}
    </div>
  );
}
