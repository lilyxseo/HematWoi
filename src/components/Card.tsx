import { PropsWithChildren } from 'react';
import clsx from 'clsx';

export default function Card({ children, className }: PropsWithChildren<{ className?: string }>) {
  return <div className={clsx('bg-white dark:bg-gray-800 rounded-2xl shadow p-4', className)}>{children}</div>;
}
