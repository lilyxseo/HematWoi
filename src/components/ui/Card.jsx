import React from "react";
export function Card({ className = "", children }) {
  return (
    <section
      className={`card ${className}`}
      style={{ containerType: "inline-size", containerName: "card" }}
    >
      {children}
    </section>
  );
}
export function CardHeader({ title, subtitle, extra }) {
  return (
    <header
      style={{
        display: "flex",
        alignItems: "center",
        justifyContent: "space-between",
        gap: "12px",
        flexWrap: "wrap",
      }}
    >
      <div>
        {title && <div className="card-title">{title}</div>}
        {subtitle && <div className="card-sub">{subtitle}</div>}
      </div>
      {extra && (
        <div
          className="actions"
          style={{ display: "flex", gap: "8px", flexWrap: "wrap" }}
        >
          {extra}
        </div>
      )}
    </header>
  );
}
export function CardBody({ children }) {
  return <div>{children}</div>;
}
export function CardFooter({ children }) {
  return <footer style={{ marginTop: "auto" }}>{children}</footer>;
}
export default { Card, CardHeader, CardBody, CardFooter };
