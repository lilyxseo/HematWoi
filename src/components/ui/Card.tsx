import { HTMLAttributes } from 'react';
import { cn } from '../../lib/utils';

export const Card = ({ className, ...props }: HTMLAttributes<HTMLDivElement>) => {
  return (
    <div
      className={cn(
        'rounded-2xl border border-slate-200 bg-white p-4 shadow-sm md:p-6',
        className
      )}
      {...props}
    />
  );
};
