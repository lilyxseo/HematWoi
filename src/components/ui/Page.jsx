import React from "react";
export function Page({ title, actions, children }) {
  return (
    <div className="container page">
      <div className="page-header">
        <h1 style={{ fontSize: "var(--fs-700)", margin: 0 }}>{title}</h1>
        <div style={{ display: "flex", gap: "8px", flexWrap: "wrap" }}>
          {actions}
        </div>
      </div>
      {children}
    </div>
  );
}
export default { Page };
