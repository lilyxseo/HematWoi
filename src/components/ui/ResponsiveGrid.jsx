import React from "react";
export default function ResponsiveGrid({ children }) {
  return <div className="card-grid">{children}</div>;
}
