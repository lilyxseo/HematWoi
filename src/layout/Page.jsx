/**
 * Page container enforcing global vertical rhythm.
 * Applies top and bottom padding using --page-y.
 */
export default function Page({ children }) {
  return (
    <main
      className="mx-auto w-full min-w-0 px-4 sm:px-6 lg:max-w-6xl lg:px-8 xl:max-w-7xl xl:px-10 2xl:max-w-[88rem] 2xl:px-12"
      style={{ paddingTop: "var(--page-y)", paddingBottom: "var(--page-y)" }}
    >
      {children}
    </main>
  );
}
