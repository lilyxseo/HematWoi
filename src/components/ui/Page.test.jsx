/* @vitest-environment jsdom */
import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/vitest';
import { Page } from './Page';
import ResponsiveGrid from './ResponsiveGrid';
import { Card, CardBody } from './Card';

describe('Page component', () => {
  it('renders layout', () => {
    const { asFragment, getByText } = render(
      <Page title="Demo" actions={<button>Action</button>}>
        <ResponsiveGrid>
          <Card>
            <CardBody>Item</CardBody>
          </Card>
        </ResponsiveGrid>
      </Page>
    );
    expect(getByText('Demo')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
