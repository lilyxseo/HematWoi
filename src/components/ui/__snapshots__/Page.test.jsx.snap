// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Page component > renders layout 1`] = `
<DocumentFragment>
  <div
    class="container page"
  >
    <div
      class="page-header"
    >
      <h1
        style="font-size: var(--fs-700); margin: 0px;"
      >
        Demo
      </h1>
      <div
        style="display: flex; gap: 8px; flex-wrap: wrap;"
      >
        <button>
          Action
        </button>
      </div>
    </div>
    <div
      class="card-grid"
    >
      <section
        class="card "
        style="container-type: inline-size; container-name: card;"
      >
        <div>
          Item
        </div>
      </section>
    </div>
  </div>
</DocumentFragment>
`;
